lib/ht/search_client/test.rb

```diff
@@ -89,6 +102,14 @@ def stub_request(endpoint, params)
 
       WebMock.stub_request(:get, service_uri.to_s)
     end
+
+    def stub_post_request(endpoint, params)
```

Comment:
I doesn't necessarily need `with` but it narrows down the scoping for the stub.
On a get you want `http://example.com/endpoint?param1=value`, on the post we are passing the payload in the request body on a different url (`http://example.com/endpoint`, no query strings), and `with` narrows down the stubbing.